src/index.js

```diff
@@ -46,47 +44,58 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^[_a-zA-Z][a-zA-Z0-9]*$|^[A-Z][A-Z]+_[A-Z][A-Z]+$'],
```

Comment:
If those examples are valid upper case notation, I think it's good to accept them in this rule (they might not be the best choice within the code, but that's a different matter). I see this enabling some better examples such as `OPTION_1` or `N_TIMES`.

Anyway, the specific regex I shared is more to illustrate the point than to be prescriptive. I think that at least supporting 0-n underscores and numbers would be important.